firewood/src/merkle.rs

```diff
@@ -1989,4 +1962,216 @@ mod tests {
             merkle.insert(key, val, root).unwrap()
         }
     }
+
+    #[test]
+    fn overwrite_leaf() {
```

Comment:
It's unclear what this test has to do with the above changes.

Reply:
It's an extra test case that wasn't previously covered.